packages/bot-web-ui/src/stores/google-drive-store.ts

```diff
@@ -119,8 +119,31 @@ export default class GoogleDriveStore {
             const current_epoch_time = Math.floor(Date.now() / 1000);
             if (current_epoch_time > Number(expiry_time)) {
                 try {
-                    //request new access token if invalid
-                    await this.client.requestAccessToken({ prompt: '' });
+                    // Remove any old tokens from localStorage
+                    localStorage.removeItem('google_access_token_expiry');
+                    localStorage.removeItem('google_access_token');
+                    // Make the token request
+                    const tokenRequest = this.client.requestAccessToken({ prompt: '' });
+
+                    const waitForToken = new Promise((resolve, reject) => {
+                        const interval = setInterval(() => {
+                            const token = localStorage.getItem('google_access_token');
+                            if (token) {
+                                clearInterval(interval);
+                                resolve(token);
+                            }
+                        }, 10); // Check every 10ms
+
+                        setTimeout(() => {
+                            clearInterval(interval); // Clear the interval after timeout
+                            reject(new Error('Token not found within timeout.'));
+                        }, 10000);
+                    });
+                    await tokenRequest;
+                    // Wait for the token to appear in localStorage
+                    await waitForToken;
+
+                    return 'verified';
                 } catch (error) {
```

Comment:
Previously, we were not waiting for the local storage update. As a result, even though we updated the google_access_token, we faced a 403 error. We need to wait until the local storage is updated.

Reply:
@mayuran-deriv what function exactly is setting the 'google_access_token' in local storage? Why do we have to run intervals to check the local storage? Is there any way to just wait for the async function that is setting the local storage?

Reply:
@shafin-deriv as i checked its coming from google server , we don't have a way to check ,
but will confirm with @rupato-deriv

Reply:
`this.client.requestAccessToken({ prompt: '' })` what's the response of this request?

Reply:
unfortunately its undefined.